### PR TITLE
feat: implement support for more well-known-types

### PR DESCRIPTION
**Pull request by odsod**
- google.protobuf.Struct (mapped to a JSON STRING)
- google.type.TimeOfDay (mapped to TIME)
- google.type.Date (mapped to DATE)
- google.type.LatLng (mapped to GEOGRAPHY)

The reason for using GEOGRAPHY as default for LatLng is because of the
clustering opportunities for tables that store locations as GEOGRAPHY.

Future SchemaOptions may override to store LatLng as a record instead.

Integration tested against public BigQuery datasets. The intention is to
identify datasets that exercise the features of this library, then
perform a load/marshal/unmarshal sequence and validate that the original
result is preserved.

### CODE REVIEW

**Comment by alethenorio** [comment]
Why not map proto struct to a BigQuery struct as well (I think they call it a record in Go)?

**Comment by odsod** [comment]
@alethenorio google.protobuf.Struct is an untyped struct without a schema - can't be mapped 1:1 with a BigQuery RECORD.

**Comment by alethenorio** [comment]
Ah I see, it's just a map. 👍

File: encoding/protobq/integration_test.go [hunk 1]
```diff
@@ -17,65 +17,111 @@ import (
 )
 
 func Test_Integration_PublicDataSets(t *testing.T) {
+	t.Parallel()
```
[the remark this answers is not in the document]
Comment (odsod) [reply encoding/protobq/integration_test.go:20]: Each instance of the test takes ~5s to run depending on connection time to BigQuery - some datasets seem to be rate limited.

The latest version of GolangCI-Lint comes with a linter for `t.Parallel()`. Do you think we should disable it? I think it's good that it makes tests run faster, and that it enforces to not share resources between tests.

[the remark this answers is not in the document]
Comment (odsod) [reply encoding/protobq/integration_test.go:20]: https://github.com/golangci/golangci-lint/pull/1503

[the remark this answers is not in the document]
Comment (alethenorio) [reply encoding/protobq/integration_test.go:20]: From the docs I understand the linter will enforce bad usage of t.Parallel so I see no reason to disable it.

I think we can make educated uses of t.Parallel and this sounds like one of them given the nature of the integration tests, just need to be careful, specially in integration tests and sharing of resources (such as a database) is more prone to happen.

**Review by alethenorio — APPROVED** [review]
Very nice work. Keep it up

File: encoding/protobq/integration_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (odsod) [reply encoding/protobq/integration_test.go:20]: Sounds good to me!

**Comment by odsod** [comment]
@alethenorio Thanks! :raised_hands:

**Pull request merged**